apps/preview/nuxt/pages/showcases/Combobox/ComboboxBasic.vue

```diff
@@ -0,0 +1,263 @@
+  SfIconExpandMore,
+  SfSwitch,
+  useDisclosure,
+  useDropdown,
+  useTrapFocus,
+  useId,
+} from '@storefront-ui/vue';
+
+const inputModel = ref('');
+const inputRef = ref();
+const dropdownRef = ref();
+const isLoadingSnippets = ref(false);
+const id = useId();
+const listId = useId();
+const isDisabled = ref(false);
+const snippets = ref<{ label: string; value: string }[]>([]);
+const isValid = ref<boolean | undefined>(undefined);
+const isSelected = ref(false);
+const { isOpen, close, open, toggle } = useDisclosure();
+const { referenceRef, floatingRef, style } = useDropdown({
+  isOpen,
+  onClose: close,
+  placement: 'bottom-start',
+  middleware: [offset(4)],
+});
+const { current: currentFocus, focusables: focusableElements } = useTrapFocus(dropdownRef, {
+  trapTabs: false,
+  arrowKeysOn: true,
+  activeState: isOpen,
+  initialFocus: false,
+});
+
+const handleFocusInput = () => {
+  const inputEl = unrefElement(inputRef)?.querySelector('input');
+  inputEl?.focus();
+};
+
+const handleReset = () => {
+  inputModel.value = '';
+  snippets.value = [];
+  close();
+  handleFocusInput();
+};
+
+const handleBlur = () => {
+  isValid.value = !!options.find((option) => option.value.toLowerCase() === inputModel.value.toLowerCase());
+};
+
+const handleInputKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
+  if (event.key === 'Escape') handleReset();
+  if (event.key === 'Enter') close();
+  if (event.key === 'ArrowUp') {
+    open();
+    if (isOpen) {
+      const focusableElementsAmount = focusableElements.value.length;
+      focusableElements.value[focusableElementsAmount - 1].focus();
+    }
+  }
+  if (event.key === 'ArrowDown') {
+    open();
+    if (isOpen) {
+      focusableElements.value[0].focus();
+    }
+  }
+};
+
+const selectOption = (phrase: string) => {
+  inputModel.value = phrase;
+  isSelected.value = true;
+  close();
+  handleFocusInput();
+};
+
+watch(inputModel, () => {
+  if (inputModel.value === '') {
+    handleReset();
+  }
+});
+
+watchDebounced(
+  inputModel,
+  () => {
+    if (inputModel.value && isSelected.value === false) {
+      const getSnippets = async () => {
+        if (!isOpen.value) open();
+        isLoadingSnippets.value = true;
+        try {
+          const data = await mockAutocompleteRequest(inputModel.value);
+          snippets.value = data;
+        } catch (error) {
+          close();
+          console.error(error);
+        }
+        isLoadingSnippets.value = false;
+      };
+      getSnippets();
+    } else {
+      isSelected.value = false;
+    }
+  },
+  { debounce: 500 },
+);
+
+type SelectOption = {
+  label: string;
+  value: string;
+};
+const options: SelectOption[] = [
+  {
+    label: 'Afghanistan',
+    value: 'afghanistan',
+  },
+  {
+    label: 'Bahamas',
+    value: 'bahamas',
+  },
+  {
+    label: 'Canada',
+    value: 'canada',
+  },
+  {
+    label: 'Denmark',
+    value: 'denmark',
+  },
+  {
+    label: 'Egypt',
+    value: 'egypt',
+  },
+];
```

Comment:
We should use some longer list of countries here. It would be best to add at least 15 here: `Albania`, `Algeria`, `Andorra`, `Angola`, "Armenia", "Antarctica", "Argentina", "American Samoa", "Austria", "Australia".

Reply:
If it's necessary now I'll add it, but I thought it was supposed to be added in the separate task with a [scrollable dropdown.](https://vsf.atlassian.net/browse/SFUI2-1193)